src/feeds.js

```diff
@@ -369,66 +369,41 @@ feeds.fetchEventsFromCalendar_ = function(feed, callback) {
   });
 };
 
-
 /**
  * Updates the notification alarms
  */
 feeds.updateNotification = function() {
   if (!options.get(options.Options.SHOW_NOTIFICATIONS)) {
     return;
   }
+  // If event deleted, then delete alarm
+  chrome.alarms.clearAll();
 
-  chrome.alarms.getAll(function(alarms) {
-    // If event deleted, then delete alarm
-    var isAlarmEventFound = false;
-    for(var i = 0; i < alarms.length; i++) {
-      isAlarmEventFound = false;
-      for(var j = 0; j < feeds.events.length; j++) {
-        if(alarms[i].name === feeds.events[j].event_id) {
-          isAlarmEventFound = true;
-          break;
-        }
-      }
-      // The event is deleted, delete alarm
-      if(!isAlarmEventFound) {
-        var alarmName = alarms[i].name;
-        chrome.alarms.clear(alarmName);
-      }
+  // Check against one event at the time
+  for (var i = 0; i < feeds.events.length; i++) {
+    if (feeds.events[i].reminders.length === 0) {
+      continue;
     }
-
-    // Check against one event at the time
-    for (var i = 0; i < feeds.events.length; i++) {
-      if (feeds.events[i].reminders.length === 0) {
+    for (var j = 0; j < feeds.events[i].reminders.length; j++) {
+      // Only create notifications of popup types
+      if (feeds.events[i].reminders[j].method !== 'popup') {
         continue;
       }
 
       // Check if the event has an alarm. This also saves the index of the alarm
-      var alarmIndex = -1;
-      var hasEventAnAlarm = alarms.some(function(alarm, index) {
-        alarmIndex = index;
-        return feeds.events[i].event_id === alarm.name;
-      });
+      var timeUntilReminderMinutes = feeds.events[i].reminders[j].minutes;
+      var eventId = feeds.events[i].event_id + ':reminder:' + timeUntilReminderMinutes;
 
-      var reminderMinutes = feeds.events[i].reminders[0].minutes;
-      var alarmSchedule = moment(feeds.events[i].start).subtract(reminderMinutes, 'minutes');
+      var alarmSchedule =
+          moment(feeds.events[i].start).subtract(timeUntilReminderMinutes, 'minutes');
       // Cancel if reminder has passed
```

Comment:
OK to skip comments like this, because the method name itself explains what is going on.